lib/index.js

```diff
@@ -95,11 +95,15 @@ angular
                         elementBottom = offsetTop(lastChoice) - containerTopOffset + height(lastChoice);
 
                         var remaining = elementBottom - containerBottom,
-                            shouldScroll = remaining <= height(container) * scrollDistance + 1;
+                            shouldScroll = remaining <= height(container) * (scrollDistance + 1);
 
                         if (shouldScroll) {
                             scope.$apply(function() {
-                                $parse(attrs['reachInfinity'])(scope);
+                                $q($parse(attrs['reachInfinity'])(scope))
```

Comment:
I've wrapped this in a promise, since the handler might return a promise (if fetching from the api). If not this ensures that there are no errors